### PR TITLE
MohammadH/ Academy TS Errors

### PR DESCRIPTION
**Pull request by mohammad-rework**
Changes:

-   ...

## Type of change

-   [x] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [x] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/pages/academy/components/video-banner/_DBanner.tsx [hunk 1]
```diff
@@ -129,10 +129,6 @@ const Dbanner = ({ featured_video_list_data, non_featured_video_list_data }: Vid
                         <GatsbyImage
                             image={getImage(video_thumbnail.imageFile)}
                             alt={thumbnail_img}
-                            width="100%"
-                            height="100%"
-                            layout="fullWidth"
-                            transformOptions={{ fit: 'cover', cropFocus: 'attention' }}
                         />
```
Comment (NikitK-deriv) [inline src/pages/academy/components/video-banner/_DBanner.tsx:132]: Do we have the same size and image behavior after this removed code?

File: src/pages/academy/components/video-banner/_VideoCarousel.tsx [hunk 2]
```diff
@@ -153,8 +151,6 @@ const VideoCarousel = ({ carousel_items }: VideoCarouselProps) => {
                                                     .gatsbyImageData
                                             }
                                             alt={thumbnail_img_alt}
-                                            width="100%"
-                                            height="100%"
```
Comment (NikitK-deriv) [inline src/pages/academy/components/video-banner/_VideoCarousel.tsx:157]: I meant include this in last comment

File: src/pages/academy/components/video-banner/_DBanner.tsx (hunk 1 above)
Comment (mohammad-rework) [reply src/pages/academy/components/video-banner/_DBanner.tsx:132]: Yes, actually `<GatsbyImage />` doesn't accept any of those properties.

File: src/pages/academy/components/video-banner/_VideoCarousel.tsx (hunk 2 above)
Comment (mohammad-rework) [reply src/pages/academy/components/video-banner/_VideoCarousel.tsx:157]: Still the same.

**Review by NikitK-deriv — DISMISSED** [review]

**Review by prince-deriv — APPROVED** [review]

File: src/pages/academy/components/search-banner/_search-banner.tsx [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+    const [hide_mobile_topic, setHideMobileTopic] = useState(false)
+    const [blog_post_url, setBlogPostURL] = useState(false)
+    const { is_eu, is_uk } = getCountryRule()
+
+    // Filter out restricted categories from the combined filter type array based on geolocation
+    useEffect(() => {
+        combined_filter_type.forEach((type) => {
+            type.items = type.items.filter((obj) => {
+                if (is_eu) return obj.is_visible_eu
+                if (is_uk) return obj.is_visible_uk
```
Comment (prince-deriv) [inline src/pages/academy/components/search-banner/_search-banner.tsx:87]: Can we link this somehow to @NikitK-deriv 's filter?

**Pull request closed without merging**